test/jdk/java/awt/Mouse/MouseModifiersUnitTest/ExtraButtonDrag.java

```diff
@@ -22,110 +22,120 @@
+        for (int i = 0; i < MouseInfo.getNumberOfButtons(); i++) {
+            buttonMask[i] = InputEvent.getMaskForButton(i + 1);
+        }
 
-                try {
-                    dragMouse(buttonMask[i], centerFrame.x, centerFrame.y, outboundsFrame.x, outboundsFrame.y);
-                } catch (IllegalArgumentException e){
-                    throw new RuntimeException("Test failed. Exception occured.", e);
-                }
+        EventQueue.invokeAndWait(() -> {
+            Point location = frame.getLocationOnScreen();
+            Dimension size = frame.getSize();
+            centerFrame = new Point(location.x + size.width / 2,
+                location.y + size.height / 2);
+            outboundsFrame = new Point(location.x + size.width * 3 / 2,
+                location.y + size.height / 2);
+        });
+
+        System.out.println("areExtraMouseButtonsEnabled() == "
+            + Toolkit.getDefaultToolkit().areExtraMouseButtonsEnabled());
+
+        for (int i = 0; i < MouseInfo.getNumberOfButtons(); i++) {
+            System.out.println("button to drag = " + (i + 1)
+                + " : value passed to robot = " + buttonMask[i]);
+
+            try {
+                dragMouse(buttonMask[i], centerFrame.x, centerFrame.y,
+                    outboundsFrame.x, outboundsFrame.y);
+            } catch (IllegalArgumentException e) {
+                throw new RuntimeException("Test failed. Exception occured.",
+                    e);
+            }
 
-                robot.delay(500);
-                //this is a choice-case for X protocol issue: native events from extra buttons doesn't contain
-                // the correct state so it's unable to decide if there is a drag or move. By default we send MOVED event.
-                //XToolkit: extra buttons should report MOVED events only
-                //WToolkit: extra buttons should report DRAGGED events only
-                if (i > 2){ //extra buttons only
-                    if (tk.equals("sun.awt.X11.XToolkit")) {
-                        if (!moved || dragged) {
-                            throw new RuntimeException("Test failed."+ tk +" Button = " +(i+1) + " moved = "+moved +" : dragged = " +dragged);
-                        }
-                    } else { //WToolkit
-                        if (moved || !dragged) {
-                            throw new RuntimeException("Test failed."+ tk +" Button = " +(i+1) + " moved = "+moved +" : dragged = " +dragged);
-                        }
+            // this is a choice-case for X protocol issue: native events from
+            // extra buttons doesn't contain
+            // the correct state so it's unable to decide if there is a drag or
+            // move. By default we send MOVED event.
+            // XToolkit: extra buttons should report MOVED events only
+            // WToolkit: extra buttons should report DRAGGED events only
```

Comment:
One suggestion could be to use `Toolkit.getDefaultToolkit().getClass().getName()` to access the toolkit names. Some other tests in the test directory use this already. However, as Phil mentioned, I suppose this current way is fine.